examples/pennylane/5_tracking.ipynb

```diff
@@ -0,0 +1,647 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 7,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "{'executions': 5, 'batches': 2, 'batch_len': 5}\n"
+     ]
+    }
+   ],
+   "source": [
+    "print(tracker.totals)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Note that `None` is not numeric, which is why the shots are not accumulated in `tracker.totals` in the previous example.\n",
+    "If we instead compute the gradient with a finite number of shots, we will see the total shots recorded.\n",
+    "Let's try the gradient with `shots=100`.\n",
+    "Since there are 5 circuit evaluations, we should expect the total shots to be `500`. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 8,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "{'executions': 5, 'shots': 500, 'batches': 2, 'batch_len': 5}\n"
+     ]
+    }
+   ],
+   "source": [
+    "with qml.Tracker(dev) as tracker:\n",
+    "    qml.grad(qnode_local)(params, shots=100)\n",
+    "print(tracker.totals)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We can also inspect the most recently recorded data with `tracker.latest`.\n",
+    "Note that not every field will be present in every update.\n",
+    "The latest update in this gradient calculation only included `'batches'` and `'batch_len'`."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 9,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "{'batches': 1, 'batch_len': 4}\n"
+     ]
+    }
+   ],
+   "source": [
+    "print(tracker.latest)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Lastly, a callback function may also be given to the device tracker.\n",
+    "This function is called each time the information in the tracker is updated.\n",
+    "This interface could be used for monitoring large batches of executions.\n",
+    "For example, let's print out the running total number of executions every time the tracker is updated."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 10,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Total executions: 1\n",
+      "Total executions: 1\n",
+      "Total executions: 2\n",
+      "Total executions: 3\n",
+      "Total executions: 4\n",
+      "Total executions: 5\n",
+      "Total executions: 5\n"
+     ]
+    }
+   ],
+   "source": [
+    "def log_executions(totals, history, latest):\n",
+    "    print(\"Total executions:\",totals[\"executions\"])\n",
+    "\n",
+    "with qml.Tracker(dev, callback=log_executions) as tracker:\n",
+    "    qml.grad(qnode_local)(params)"
+   ]
+  },
```

Comment:
Overall the The PennyLane Device Resource Tracker section is a bit too long before getting to the meat of the demo (i.e. Using the Tracker to Limit Resource Usage). I'd suggest weaving these explanations of the usage and functionality of the Device Resource Tracker into the Using the Tracker to Limit Resource Usage section, basically introducing and explaining them as they arise in the code.

This will help the notebook as a whole feel more motivated, guiding the reader through why they should be interested in each feature by way of an example.